redis/src/main/scala/zio/redis/ByteStream.scala

```diff
@@ -28,57 +27,59 @@ import java.nio.channels.{AsynchronousSocketChannel, Channel, CompletionHandler}
 private[redis] object ByteStream {
   trait Service {
     def read: Stream[IOException, Byte]
-    def write(chunk: Chunk[Byte]): IO[IOException, Unit]
+    def write(chunk: Chunk[Byte]): IO[IOException, Option[Unit]]
```

Comment:
What's the purpose of option?

Reply:
Since Option is required for matching here(case State.Start), this is one of the areas I find most confusing.
```scala
  private[redis] final val Decoder: ZPipeline[Any, RedisError.ProtocolError, Byte, Option[RespValue]] = {
    val Sinker: ZSink[Any, RedisError.ProtocolError, String, String, Option[RespValue]] = {
      import internal.State
      ZSink.fold[String, State](State.Start)(_.inProgress)(_ feed _).mapZIO {
        case State.Done(value) => ZIO.succeedNow(Some(value))
        case State.Failed      => ZIO.fail(RedisError.ProtocolError("Invalid data received."))
        // ZSink fold will return a State.Start when contFn is false
        case State.Start => ZIO.succeedNow(None)
        case other       => ZIO.dieMessage(s"Deserialization bug, should not get $other")
      }

    }
```